File: app/src/main/java/com/erotc/learning/adapter/ResultAdapter.java
package com.erotc.learning.adapter;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.erotc.learning.R;
import com.erotc.learning.dao.DictionaryEntry;

import java.util.ArrayList;
import java.util.List;

/**
 * Created on 11/12/2018.
 */
public class ResultAdapter extends RecyclerView.Adapter<ResultAdapter.ResultViewHolder> {
    private List<DictionaryEntry> mResultList;
    private OnClickListener mOnClickListener;


    public interface OnClickListener {
        void onClick(View view);
    }

    public ResultAdapter(OnClickListener onClickListener) {
        this.mResultList = new ArrayList<>();
        this.mOnClickListener = onClickListener;
    }

    @NonNull
    @Override
    public ResultViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.layout_result, parent, false);

        view.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if ( mOnClickListener != null ){
                    mOnClickListener.onClick(view);
                }
            }
        });

        return new ResultViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ResultViewHolder holder, int position) {
        DictionaryEntry entry = mResultList.get(position);

        if ( entry != null ){
            holder.bindTo(entry);
        } else {
            holder.clear();
        }
    }

    @Override
    public int getItemCount() {
        return mResultList.size();
    }

    public void setResultList(List<DictionaryEntry> resultList) {
        this.mResultList = resultList;
    }

    public static class ResultViewHolder extends RecyclerView.ViewHolder {
        private TextView mTagalog;
        private TextView mHiligaynon;
        private TextView mIlocano;

        public ResultViewHolder(View view){
            super(view);

            mTagalog = view.findViewById(R.id.tagalog);
            mHiligaynon = view.findViewById(R.id.hiligaynon);
            mIlocano = view.findViewById(R.id.ilocano);
        }

        public void bindTo(DictionaryEntry entry){
            mTagalog.setText(entry.getTagalog());
            mHiligaynon.setText(entry.getHiligaynon());
            mIlocano.setText(entry.getIlocano());
        }

        public void clear(){
            mTagalog.setText("");
            mHiligaynon.setText("");
            mIlocano.setText("");
        }
    }
}

File: app/src/main/java/com/erotc/learning/adapter/ResultAdapter.kt
package com.erotc.learning.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.erotc.learning.R
import com.erotc.learning.adapter.ResultAdapter.ResultViewHolder
import com.erotc.learning.data.DictionaryEntry
import kotlinx.android.synthetic.main.layout_result.view.*

/**
 * Created on 11/12/2018.
 */
class ResultAdapter(private val onClickListener: OnClickListener?) : RecyclerView.Adapter<ResultViewHolder>() {
    private var resultList: List<DictionaryEntry> = arrayListOf()

    interface OnClickListener {
        fun onClick(view: View?)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ResultViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.layout_result, parent, false)
        view.setOnClickListener { view -> onClickListener?.onClick(view) }
        return ResultViewHolder(view)
    }

    override fun onBindViewHolder(holder: ResultViewHolder, position: Int) {
        val entry = resultList[position]
        holder.bindTo(entry)
    }

    override fun getItemCount(): Int {
        return resultList.size
    }

    fun setResultList(resultList: List<DictionaryEntry>) {
        this.resultList = resultList
    }

    class ResultViewHolder(private val view: View) : RecyclerView.ViewHolder(view) {
        fun bindTo(entry: DictionaryEntry) {
            view.text_tagalog.text = entry.tagalog
            view.text_hiligaynon.text = entry.hiligaynon
            view.text_ilocano.text = entry.ilocano
        }

        fun clear() {
            view.text_tagalog.text = ""
            view.text_hiligaynon.text = ""
            view.text_ilocano.text = ""
        }
    }
}